File: src/main/java/net/guizhanss/fastmachines/utils/items/SlimefunItemExt.java
package net.guizhanss.fastmachines.utils.items;

import org.bukkit.World;
import org.bukkit.inventory.ItemStack;

import io.github.thebusybiscuit.slimefun5.api.items.SlimefunItem;

import net.guizhanss.fastmachines.libs.guizhanlib.items.Items;

public final class SlimefunItemExt {

    private SlimefunItemExt() {
    }

    public static SlimefunItem getSfItem(String id) {
        return SlimefunItem.getById(id.toUpperCase());
    }

    public static boolean isDisabled(ItemStack item) {
        if (!Items.isSlimefunItem(item)) {
            return false;
        }
        return Items.getSlimefunItem(item).isDisabled();
    }

    public static boolean isDisabledIn(ItemStack item, World world) {
        if (!Items.isSlimefunItem(item)) {
            return false;
        }
        return Items.getSlimefunItem(item).isDisabledIn(world);
    }
}

File: src/main/kotlin/net/guizhanss/fastmachines/utils/items/SlimefunItemExt.kt
package net.guizhanss.fastmachines.utils.items

import io.github.thebusybiscuit.slimefun5.api.items.SlimefunItem
import net.guizhanss.fastmachines.libs.guizhanlib.items.getSlimefunItem
import net.guizhanss.fastmachines.libs.guizhanlib.items.isSlimefunItem
import org.bukkit.World
import org.bukkit.inventory.ItemStack

fun String.getSfItem(): SlimefunItem? =
    SlimefunItem.getById(this.uppercase())

fun ItemStack.isDisabled(): Boolean {
    if (!this.isSlimefunItem()) return false

    val item = this.getSlimefunItem()
    return item.isDisabled
}

fun ItemStack.isDisabledIn(world: World): Boolean {
    if (!this.isSlimefunItem()) return false

    val item = this.getSlimefunItem()
    return item.isDisabledIn(world)
}
